File: SDKs/googleapis.com/civicinfo/v2/python/src/sdk/models/shared/divisionsearchresult.py
from dataclasses import dataclass, field
from typing import List,Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclass
class DivisionSearchResult:
    r"""DivisionSearchResult
    Represents a political geographic division that matches the requested query.
    """
    
    aliases: Optional[List[str]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('aliases') }})
    name: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    ocd_id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ocdId') }})
    

File: SDKs/googleapis.com/civicinfo/v2/typescript/src/sdk/models/shared/divisionsearchresult.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// DivisionSearchResult
/** 
 * Represents a political geographic division that matches the requested query.
**/
export class DivisionSearchResult extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=aliases" })
  aliases?: string[];

  @SpeakeasyMetadata({ data: "json, name=name" })
  name?: string;

  @SpeakeasyMetadata({ data: "json, name=ocdId" })
  ocdId?: string;
}
